runtime/parachains/src/session_info.rs

```diff
@@ -113,6 +108,25 @@ pub mod pallet {
 	#[pallet::getter(fn session_executor_params)]
 	pub(crate) type SessionExecutorParams<T: Config> =
 		StorageMap<_, Identity, SessionIndex, ExecutorParams>;
+
+	/// Pending executor parameters
+	#[pallet::storage]
+	pub(crate) type PendingExecutorParams<T: Config> =
+		StorageValue<_, Option<ExecutorParams>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight((
+			Weight::from_parts(0, 0), // FIXME: How do we calculate weights?
```

Comment:
Where does the weight come from?

Reply:
https://github.com/paritytech/polkadot/pull/4938 try to follow this pr for example, or feel free to ask if you're lost